src/ann_solo/reader.py

```diff
@@ -162,22 +197,30 @@ def close(self) -> None:
             del self._parser
 
     def __enter__(self) -> 'SpectralLibraryReader':
-        self.open()
+        verify_extension(['.splib','.sptxt','.mgf'], self._filename)
+        _, ext = os.path.splitext(os.path.basename(self._filename))
+
+        if ext == '.splib':
+            self.open()
         return self
 
     def __exit__(self, exc_type, exc_value, traceback) -> None:
-        self.close()
+        verify_extension(['.splib', '.sptxt', '.mgf'], self._filename)
+        _, ext = os.path.splitext(os.path.basename(self._filename))
```

Comment:
**suggestion:** Rather than splitting off the extension every time here, it seems more logical to do this in the constructor and store the result in an internal variable.